hydra/grammar/OverrideParser.g4

```diff
@@ -30,31 +36,34 @@ element:
     | function
 ;
 
-argName: ID EQUAL;
-function: ID POPEN (argName? element (COMMA argName? element )* )? PCLOSE;
-
 simpleChoiceSweep:
       element (COMMA element)+                   // value1,value2,value3
 ;
 
+// Functions.
+
+argName: ID EQUAL;
+function: ID POPEN (argName? element (COMMA argName? element )* )? PCLOSE;
+
+// Data structures.
+
+listValue: BRACKET_OPEN sequence? BRACKET_CLOSE;                          // [], [1,2,3], [a,b,[1,2]]
+dictValue: BRACE_OPEN (keyValuePair (COMMA keyValuePair)*)? BRACE_CLOSE;  // {}, {a:10,b:20}
+keyValuePair: ID COLON element;
+sequence: element (COMMA element)*;
```

Comment:
The motivation for the small changes to list / dict definitions is related to the future addition of interpolations in the parser:

* Lists can share the sequence rule with interpolationResolver
* Writing a dict as several key-value pairs makes both the grammar and visitor code more readable IMO (especially with the future addition of interpolations into the mix)

Reply:
let's do the sequence change later. it doesn't quiet make sense in this context.

Reply:
Ok, I force pushed a new version that doesn't have this change